src/DataValues/DecimalValue.php

```diff
@@ -355,10 +355,11 @@ public function getArrayValue() {
 	 *
 	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
 	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
-	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
+	 *  newFromArrayValue). Instead, use the constructor or use DataValue builder
+	 *  callbacks. (e.g. @see DataValueDeserializer)
```

Comment:
> Also I find the typography (brackets outside of the sentence) weird.

Oh, come on. Just move the full stop to the end:
> Instead, use the constructor or use DataValue builder callbacks (e.g. @see DataValueDeserializer).

What I find better in the version @filbertkm is that it is more generic. What you suggest instead is:
 - assuming one must use some static constructor in those deserializer callbacks. Maybe it is true in the end. But far too detailed for this particular place IMO.
 - IMO saying "introduce a new X but call it Y when you need it" is a bit odd thing to say in the deprecation notice about removing "X".